jobs/pywikibot/scripts/updatestatistics.py

```diff
@@ -143,6 +142,22 @@ def update_list_of_pages(template, pages):
         template_page.save('Updated on ' +
                            date.today().strftime('%B %d, %Y'))
 
+def deduplicate_lists(list_to_dedup, base_list):
+    '''
+    will return list_to_dedup after removing any element that exists in
+    base_list
+
+    will maintain the order of elements in list_to_dedup
+    '''
+
+    deduped_list = [ ]
+    base_dict = OrderedDict.fromkeys(base_list)
```

Comment:
This can be a `set` now? I thought the way you would use this is:

```python
deduped_list = OrderedDict.fromkeys(list_to_dedup)
for elem in base_list:
  deduped_list.remove(elem) # handle missing keys appropriately
return deduped_list
```

but if we're just checking membership of the base list, we don't need a dict?

Reply:
Yeah, no need for a dict. I didn't know that set membership checking is also O(1). I have used a set here and removed the `OrderedDict` import. :sweat_drops: